packages/authentication/src/providers/auth-action.provider.ts

```diff
@@ -49,8 +49,28 @@ export class AuthenticateActionProvider implements Provider<AuthenticateFn> {
       return undefined;
     }
 
-    const userProfile = await strategy.authenticate(request);
-    if (!userProfile) {
+    const authResponse = await strategy.authenticate(request);
+    let userProfile: UserProfile;
+
+    // response from `strategy.authenticate()` could return an object of type UserProfile or RedirectRoute
+    if (RedirectRoute.isRedirectRoute(authResponse)) {
+      const redirectOptions = authResponse as RedirectRoute;
+      // set the redirect options in the request query
+      // set redirection url as 'x-loopback-authentication-redirect-url'
+      // set redirection status as 'x-loopback-authentication-redirect-status'
+      // controller will handle actual redirection
+      request.query[
+        'x-loopback-authentication-redirect-url'
+      ] = redirectOptions.getTargetLocation();
+      request.query[
+        'x-loopback-authentication-redirect-status'
+      ] = redirectOptions.getStatusCode();
```

Comment:
We should avoid mutating `request` as much as possible.

Reply:
What about the following?

```ts
requestCtx.bind('authentication.oauth2.redirectUrl').to(...);
```

Reply:
this means the request context will have to be be injected into the controller, I saw Miroslav mentioning that as an anti pattern in one of the other issues , https://github.com/strongloop/loopback-next/issues/1881#issuecomment-534596582

Reply:
ah, we dont have to inject the whole request context, we can just inject the url value alone. you are right. I will make this change.

Reply:
agreed